### DISCUSSION TITLE
SrcMac for IXP traffic sources

### DISCUSSION THREAD

**Discussion by empi89** [akvorado/akvorado#414] [Q&A]
Is there a way to add a src mac filter? That would help lot to see which IXP peer is sending how much traffic. 

As an addition it could also be great to enrich this data by arp table from snmp and ip data from peeringdb api to have fully resolved information on sending peer.

**Comment by vincentbernat** [comment]
MAC addresses would be an easy addition when #211 is resolved. As for the remaining items, this would be more work.

**Comment by empi89** [comment]
Sounds great, thanks! I think it would already help with the src address to get a first idea. The remaining will only make it more readable :-)

**Comment by vincentbernat** [comment]
It has been added here: https://github.com/akvorado/akvorado/commit/9eee46caded6fa6cc2dfb90d1941b2ef05d15e9f

**Comment by empi89** [comment]
Thanks! I added
```
inlet:
  schema:
    enabled:
      - SrcMAC
      - DstMAC
```
to my config and switched image in docker-compose.yml from :latest to :main. All services are starting fine but I was not able to choose srcmac or dstmac as filters. Is there some other config missing?

**Reply by vincentbernat** [reply]
Move `schema` outside inlet. It should be at the root. Otherwise, only inlet is aware of this change. Both the console and the orchestrator need it. Be aware there is a lot of changes when switching to main. It may be a good idea to snapshot your data.

**Comment by empi89** [comment]
Could it be that there is an issue with the datatype? I got:
akvorado-akvorado-orchestrator-1     | 2023-01-24T10:24:26.003950999Z {"level":"error","error":"cannot create flows: code: 50, message: Unknown data type family: Uint64. Maybe you meant: ['UInt64','Int64']","time":"2023-01-24T10:24:26Z","caller":"akvorado/orchestrator/clickhouse/root.go:89","module":"akvorado/orchestrator/clickhouse","message":"database migration error"}
in log and in the common/schema/definition.go other uint64 datatypes are with uppercase "I".

**Reply by vincentbernat** [reply]
Yes, you are right. I'll add this to test coverage.

**Comment by empi89** [comment]
Thanks, looks good now :-)
